### ISSUE TITLE
Accuracy on TUAB Lower Than Reported Despite Following Setup

### ISSUE THREAD

**Issue by ChenHu-ML** [wjq-learning/CBraMod#12]
**Hi,**

First of all, thank you for your outstanding work—your research and codebase have been incredibly insightful and helpful.

I’m currently trying to reproduce the reported results on TUAB using the provided code. However, I’m seeing much lower accuracy than expected.

---

###  What I Did

1. Downloaded and preprocessed the **TUAB** dataset using your official preprocessing pipeline.
2. Ran the training script `finetune_main.py` using the same parameters described in the paper. Here is the exact command I used (for TUAB):

```bash
python /CBraMod/finetune_main.py \
  --seed 3407 \
  --cuda 0 \
  --epochs 50 \
  --batch_size 64 \
  --lr 1e-4 \
  --weight_decay 5e-2 \
  --optimizer AdamW \
  --clip_value 1 \
  --dropout 0.1 \
  --downstream_dataset TUAB\
  --datasets_dir /data/EEGdata/TUAB/edf/process_cbramod \
  --num_of_classes 1 \
  --model_dir ./models_weights/ \
  --num_workers 16 \
  --label_smoothing 0.1 \
  --frozen False \
  --use_pretrained_weights True \
  --foundation_dir pretrained_weights/pretrained_weights.pth
```
---

* **I got result as below**: \~epoch1_acc_0.77637_pr_0.86583_roc_0.86059
  These are significantly lower than the results reported in your paper.

**Comment by wjq-learning** [comment]
Hi,

Thank you for your detailed description and for following all the recommended settings carefully.

### Regarding TUAB:
We're also a bit puzzled by the results — especially since the best model was found at **epoch 1**, which is unusual. In our setup, the TUAB dataset contains the following number of samples:
- Train: **301,443**
- Validation: **71,067**
- Test: **36,945**
- **Total: 409,455**

Could you please double-check whether your data split matches these numbers? A mismatch here could explain the performance discrepancy.

As a next step, you may want to try:
- Training with a **different random seed**
- Adjusting the **learning rate** (especially for the classification head when `multi_lr=True`)
- Increasing the **number of training epochs**

We’ll continue to investigate on our end as well.

### Regarding TUEV:
The preprocessing code for TUEV was adapted from the [BIOT](https://github.com/ycq091044/BIOT) repository, and we verified that our final sample counts match those reported in the BIOT paper. So it’s unlikely that the issue stems from our code.

We used **TUEV version 2.0.0**, while the latest official version is now **2.0.1**. It's possible that some changes or additions have been made by the dataset maintainers, which might explain the differences in file count and total size you're observing.

You may want to check which version of the dataset you are using.

Thanks again for your careful work and for raising these questions — we really appreciate it!

Feel free to keep us posted, and we’ll be happy to assist further.

**Comment by wjq-learning** [comment]
Thank you so much for your continued support and for sharing your detailed findings — it's really helpful for improving the codebase.

It's very encouraging to see that using an average pooling-based classification head leads to more stable training on TUAB. I truly appreciate your effort in exploring this improvement.

Regarding the dataset splits for both TUAB and TUEV: you're absolutely right. The preprocessing code used in BIOT and LaBraM contains sources of randomness (e.g., `os.listdir()` ordering and `np.random.shuffle()`), which has unfortunately been carried over into CBraMod. Since I directly adopted the results in BIOT and LaBraM without having direct access to their exact data splits, even I cannot guarantee that our experiments were conducted under the same split conditions. As a result, I also cannot ensure a fair comparison across others and CBraMod.

To fully resolve this issue, we are planning to refine the preprocessing code to eliminate all randomness in the dataset splitting process and make it fully reproducible.

Please stay tuned for our upcoming updates — I’ll do my best to provide a more stable and reliable preprocessing pipeline soon.

Thank you again for your valuable feedback and patience!
